### ISSUE TITLE
Fatal Error on startup after installing 18.03.0-ce-mac59

### ISSUE THREAD

**Issue by nsecord** [docker/for-mac#2727]
Please, check https://docs.docker.com/docker-for-mac/troubleshoot/.
Issues without logs and details cannot be debugged, and will be closed.

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have submitted Diagnostics
  - [x] I have included the Diagnostics ID in this Issue

### Expected behavior
Expected Docker to restart after downloading and installing the latest version.

### Actual behavior
After downloading and installing the latest version of Docker 
I get the following error message each time I try to restart Docker:

Fatal Error
com.docker.supervisor failed to start
Exit code 1

I tried "Reset Docker to factory defaults" but the result is the same.

### Information
Diagnostic ID : 9653915B-2579-4B7F-A6E5-351617222E35/20180326-190106

```
Docker for Mac: version: 18.03.0-ce-mac59 (dd2831d4b7421cf559a0881cc7a5fdebeb8c2b98)
macOS: version 10.13.3 (build: 17D102)
logs: /tmp/9653915B-2579-4B7F-A6E5-351617222E35/20180326-190106.tar.gz
failure: No Docker.qcow2 or Docker.raw found: the VM has never been started
[ERROR]  vpnkit
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Volumes/Macintosh_HD/Users/normansecord/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ENOENT, \"connect\", \"\")")) connecting to /Volumes/Macintosh_HD/Users/normansecord/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         vpnkit check failed with: Unix.Unix_error(Unix.ENOENT, "connect", "")
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[ERROR]  moby
         /Volumes/Macintosh_HD/Users/normansecord/Library/Containers/com.docker.docker/Data/vms/0/console-ring does not exist
[OK]     system
[OK]     moby-syslog
[ERROR]  kubernetes
#ffb2b2#         kubernetes check failed with: (Failure
   "exec: /usr/local/bin/kubectl  --context docker-for-desktop --namespace docker get pods -o go-template='{{range .items}}{{.metadata.name}}\
  \n{{end}}': exit 1")##
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         /var/run/docker.sock does not exist
         Unexpected error (No such file or directory) connecting to /var/run/docker.sock
         /Volumes/Macintosh_HD/Users/normansecord/Library/Containers/com.docker.docker/Data/s60 does not exist
         Unexpected error (No such file or directory) connecting to /Volumes/Macintosh_HD/Users/normansecord/Library/Containers/com.docker.docker/Data/s60
         docker ps failed
[ERROR]  disk
         No Docker.qcow2 or Docker.raw found: the VM has never been started
```

### Steps to reproduce the behavior

  1. Docker 17.12 was installed (and working fine) and the automatic updater announced that 18.03 was available, do I want to install it
  2. Click download and install and the error follows ...

**Comment by carmaa** [comment]
Same behavior here.

**Comment by mirestrepo** [comment]
Same here
Diagnostic Id: E88857BF-A8AD-41D7-AFA5-C801A6BF8647

# Information

Docker for Mac: version: 18.03.0-ce-mac59 (dd2831d4b7421cf559a0881cc7a5fdebeb8c2b98)
macOS: version 10.13.3 (build: 17D102)
logs: /tmp/E88857BF-A8AD-41D7-AFA5-C801A6BF8647/20180327-084840.tar.gz
failure: com.docker.vpnkit is not running
[ERROR]  vpnkit
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/isa/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/isa/Library/Containers/com.docker.docker/Data/s51
         com.docker.vpnkit is not running
         vpnkit check failed with: Unix.Unix_error(Unix.ECONNREFUSED, "connect", "")
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux is not running
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     files
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/isa/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     disk

**Comment by gtardif** [comment]
Hi @nsecord, unfortunately this is dues to the socket file name hitting a size limit, partly because your user home folder is in `/Volumes/Macintosh_HD/Users` rather that straight `/Users/... (See https://unix.stackexchange.com/a/367012/128332). 
In the current release, this folder used for vm files is not configurable, the workaround would be change your home folder location, or maybe rename `normansecord` to something shorter (you're just above the size limit for the full filename).

**Comment by gtardif** [comment]
@mirestrepo Actually looking at your diagnostics, your issue is a duplicate of #2739, we're looking into it ; "Reset to factory defaults" should fix your case, if you can do this.

**Comment by nsecord** [comment]
@gtardif: Thanks for your response.  Attached is the requested screenshot.

![screen shot 2018-03-28 at 15 33 42](https://user-images.githubusercontent.com/481897/38032504-206e92b0-329e-11e8-838c-d74072a93813.png)


This is a bit of a bizarre issue for me that I haven't run across before.  

Can you clarify for me: the path that is causing the error is $HOME/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux ?  
If I add my home path to this I get something that is 112 characters long.  According to the StackExchange issue you pointed to, the path length needs to be less than 104 characters so I need to shorten things considerably, possibly redo the whole mount point for my user directory.  That's going to take a little bit of thought and some serious backups.

Everything was running fine on 17.12.1. I was looking for a link to try to roll back to that for the moment but I haven't found one.  Any suggestions?

**Comment by nsecord** [comment]
Thank @bandrejx !!

**Comment by gtardif** [comment]
@mirestrepo Your issue is due to a problem when migrating straight from 17.09 to 18.03. 
It should be fixed with the latest Docker for Mac 18.03.0-mac60 Stable release (you can download at https://download.docker.com/mac/stable/Docker.dmg)

**Comment by akimd** [comment]
Hi people,

We believe that the Stable release we are about to make fixes the issues wrt socket name length (https://github.com/docker/pinata/pull/9241).  Could someone check that this preview actually helps you?  Thanks in advance!

https://download.docker.com/mac/stable/24214/Docker.dmg

**Comment by gtardif** [comment]
This has been fixed with Edge version 18.05.0-ce-rc1-mac63 and Stable version 18.03.1-ce-mac64.

**Comment by manodupont** [comment]
Not closed for me.

```
manueldupont@manu-macbook-perso .ssh $ docker version
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:05:26 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:13:46 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

It still gives me that window where com.docker.supervisor  has failed to start, with the three buttons Exit, Diagnose & Feedback, and Rest to factory defaults (which gives me docker back)

Im on High Sierra 10.13.5
